src/main/java/core/basesyntax/figures/figurehandlers/FigureSupplier.java

```diff
@@ -0,0 +1,88 @@
+package core.basesyntax.figures.figurehandlers;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURES_COUNT = 5;
+    private static final int RANDOM_VALUES_BORDER = 30;
+    private final Figure defaultFigure = new Circle("White", 10);
+
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(FIGURES_COUNT);
+
+        switch (randomIndex) {
+            case 0: {
+                String color = colorSupplier.getRandomColor();
+                int radius = getRandomValue();
+                return getRandomCircle(color, radius);
+            }
+            case 1: {
+                String color = colorSupplier.getRandomColor();
+                int lowBase = getRandomValue();
+                int highBase = getRandomValue();
+                int leg = getRandomValue();
+                int height = getRandomValue();
+                return getRandomIsoscelesTrapezoid(color, lowBase, highBase, leg, height);
+            }
+            case 2: {
+                String color = colorSupplier.getRandomColor();
+                int hypotenuse = getRandomValue();
+                int firstLeg = getRandomValue();
+                int secondLeg = getRandomValue();
+                return new RightTriangle(color, hypotenuse, firstLeg, secondLeg);
+            }
+            case 3: {
+                String color = colorSupplier.getRandomColor();
+                int firstSide = getRandomValue();
+                int secondSide = getRandomValue();
+                return new Rectangle(color, firstSide, secondSide);
+            }
+            case 4: {
+                String color = colorSupplier.getRandomColor();
+                int side = getRandomValue();
+                return new Square(color, side);
+            }
+            default:
+                return getDefaultFigure();
```

Comment:
you create a color in each case, I would bring this field to the beginning and use it in each case. I also recommend not creating a variable for each random field and calling it immediately in the method parameters